src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -191,6 +191,11 @@ internal static bool EnhancedAnnotationsEnabled(IExecutionContext context)
         {
             return context.Global.Variables.GetBoolean("DistributedTask.EnhancedAnnotations") ?? false;
         }
+
+        internal static bool StepSummaryEnabled(IExecutionContext context)
```

Comment:
i would move this into `SetStepSummaryCommand` and leave `ActionCommandManager.cs` unchange.